mbtools/models.py

```diff
@@ -28,69 +36,115 @@ def __init__(self, mbz_path):
         self.mbz_path = Path(mbz_path)
-        self.activity_path = Path(activity_path)
-        self.activity_type = activity_type
-        self.activity_filename = \
-            str(self.activity_path / f"{self.activity_type}.xml")
+        self.activity_path = activity_path
+        self.activity_filename = str(self.activity_path / "lesson.xml")
         self.etree = etree.parse(self.activity_filename)
 
     def html_elements(self):
         elements = []
-        if self.activity_type == "lesson":
-            xpath_query = "//pages/page/contents"
-            elements = self.etree.xpath(xpath_query)
-            xpath_query = "//pages/page/answers/answer_text"
-            elements.extend(self.etree.xpath(xpath_query))
-            return [MoodleHtmlElement(el) for el in elements]
-        elif self.activity_type == "page":
-            xpath_query = "//page/content"
-            elements = self.etree.xpath(xpath_query)
-            return [MoodleHtmlElement(el) for el in elements]
-        elif self.activity_type == "quiz":
-            xpath_query = \
-                "//quiz/question_instances/question_instance/questionid"
-            ids = []
-            for question in self.etree.xpath(xpath_query):
-                ids.append(question.text)
-
-            question_objs = \
-                MoodleQuestionBank(self.mbz_path).questions(ids)
-            elements = []
-            for question in question_objs:
-                elements.extend(question.html_elements())
-            return elements
-        else:
-            return []
+        lesson_name = self.etree.xpath("//name")[0].text
+        pages = self.etree.xpath("//pages/page")
+        for page in pages:
+            page_id = page.attrib["id"]
+            page_title = page.xpath("title")[0].text
+            contents = page.xpath("contents")
+
+            location = \
+                f'Lesson: {lesson_name} Page_id={page_id}'\
+                f'Page Title: {page_title}'
+            elements.append(MoodleHtmlElement(contents[0], location))
+
+            if len(page.xpath("answers")) > 0:
+                answer_texts = page.xpath("answers/answer_text")
+                for answer in answer_texts:
+                    answer_location = f'{location} Answer: {answer.text}'
+                    elements.append(
+                        MoodleHtmlElement(answer, answer_location))
+        return elements
+
+
+class MoodlePage:
+    def __init__(self, activity_path, mbz_path):
+        self.mbz_path = Path(mbz_path)
+        self.activity_path = activity_path
+        self.activity_filename = str(self.activity_path / "page.xml")
+        self.etree = etree.parse(self.activity_filename)
+
+    def html_elements(self):
+        xpath_query = "//page/content"
+        name = self.etree.xpath("//page/name")[0].text
+        elements = self.etree.xpath(xpath_query)
+        return [MoodleHtmlElement(el, name) for el in elements]
```

Comment:
Just a quick question / sanity check: Does this not need to pass a location in the constructor?

Reply:
☝️ Right now, when a page is a subelement of a lesson, we are not creating a separate `MoodlePage` element for it. We only create page elements when they exist as separate files pointed to from moodle_backup.xml.

Since MoodlePage objects always only represent page.xml files - items that appear as peers to lessons - their location relative to everything else is the same as a lesson or quiz: "In this course". None of these three (Lesson, Quiz, Page) objects need a location passed in because its given that they exist as immediate children of the course itself. 

Creating separate `MoodlePage` Elements every time we see a page in a lesson seems to me like unnecessary complication towards our final goal in the `MoodleLesson` class, which is parsing and understanding the html contents within their contexts. So long as we can tell exactly where an html snipped is located and the details that make it function correctly, I don't see a reason to create more structures - But i may be missing a future use case ! 🤷‍♂️

Reply:
With fresh :eyes: I see now that you are passing a location (the `name`). I just got confused when reading that the first time.